test/lazy/test_kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+import unittest
+
+import torch
+
+from gpytorch.lazy import DiagLazyTensor, KroneckerProductAddedDiagLazyTensor, KroneckerProductLazyTensor, NonLazyTensor
+from gpytorch.test.lazy_tensor_test_case import LazyTensorTestCase
+
+
+class TestKroneckerProductAddedDiagLazyTensor(unittest.TestCase, LazyTensorTestCase):
+    # this lazy tensor has an explicit inverse so we don't need to run these
```

Comment:
should we add a small unit tests that checks that this actually uses the right methods / doesn't use CG / Cholesky?

Reply:
I think those are mostly dealt with by the `should_call_lanczos` and `should_call_cg` flags. Cholesky might be a good thing to check though.

Reply:
Just added a `inv_matmul` style unit test to double check if the cholesky is called.